### PR TITLE
[APPS-4621] Update messages shown by the SyncAPI SDK

### PR DESCRIPTION
**Pull request by ghost**
- Developer(s): @kopylets

- Ticket: https://spryker.atlassian.net/browse/APPS-4621

- Release Group: URL_HERE

- merge: merge

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   SyncApi               | patch (currently 0.x)  |                       |

-----------------------------------------

### Improvements

- The validation/generation error messages have been updated to be more verbose and aligned.

### CODE REVIEW

File: src/SprykerSdk/SyncApi/Message/SyncApiError.php [hunk 1]
```diff
@@ -14,23 +14,39 @@ class SyncApiError
      */
     public static function couldNotGenerateCodeFromOpenApi(): string
     {
-        return static::format('Could not generate code from Open API schema file. Schema file is not valid, please run validation before generating code.');
+        return static::format(
+            'SyncAPI schema generation error: Could not generate code from Open API schema file. Schema file is not valid, please run validation before generating code.',
```
Comment (stereomon) [inline src/SprykerSdk/SyncApi/Message/SyncApiError.php]: Can you pass the path here as well?

File: src/SprykerSdk/SyncApi/OpenApi/Builder/OpenApiCodeBuilder.php [hunk 2]
```diff
@@ -140,7 +140,11 @@ protected function generateResourceMethodResponse(
         $resourceHttpMethodsWithHttpResponseCodes = [];
 
         if (!isset($openApi->paths) || empty($openApi->paths)) {
-            $this->openApiResponseTransfer->addError($this->messageBuilder->buildMessage(SyncApiError::openApiDoesNotDefineAnyPath()));
+            $this->openApiResponseTransfer->addError(
+                $this->messageBuilder->buildMessage(
+                    SyncApiError::openApiDoesNotDefineAnyPath($openApiRequestTransfer->getTargetFile()),
```
Comment (stereomon) [inline src/SprykerSdk/SyncApi/OpenApi/Builder/OpenApiCodeBuilder.php]: Use `OrFail()` method.

File: src/SprykerSdk/SyncApi/Message/SyncApiError.php (hunk 1 above)
Comment (stereomon) [reply src/SprykerSdk/SyncApi/Message/SyncApiError.php]: Please extract a const for `SyncAPI schema generation error:`.

File: src/SprykerSdk/SyncApi/Message/SyncApiError.php [hunk 3]
```diff
@@ -14,23 +14,39 @@ class SyncApiError
...
+     *
      * @return string
      */
-    public static function openApiDoesNotDefineAnyPath(): string
+    public static function openApiDoesNotDefineAnyPath(string $path): string
     {
-        return static::format('Couldn\'t find any path definition. You need at least one.');
+        return static::format(
+            sprintf(
+                'SyncAPI schema validation error: Couldn\'t find any path definition in your "%s" schema file. Please update the schema file.',
```
Comment (stereomon) [inline src/SprykerSdk/SyncApi/Message/SyncApiError.php]: Please extract a const for `SyncAPI schema validation error:`.

File: src/SprykerSdk/SyncApi/Message/SyncApiError.php [hunk 4]
```diff
@@ -101,7 +141,9 @@ public static function couldNotParseOpenApi(string $path): string
      */
     public static function couldNotFinOpenApi(string $path): string
```
Comment (stereomon) [inline src/SprykerSdk/SyncApi/Message/SyncApiError.php]:
```suggestion
    public static function couldNotFindOpenApi(string $path): string
```

File: tests/_support/Helper/OpenApiValidatorHelper.php [hunk 5]
```diff
@@ -68,11 +68,37 @@ public function haveOpenApiFileWithPathButInvalidHttpMethod(): void
...
+        return 'resources/api';
+    }
+
+    /**
+     * @param string $path
+     * @param array $files
+     *
+     * @return array
+     */
+    protected function buildStructureByPath(string $path, array $files): array
```
Comment (stereomon) [inline tests/_support/Helper/OpenApiValidatorHelper.php:90]: Why this change?

**Review by stereomon — CHANGES_REQUESTED** [review]

**Comment by stereomon** [comment]
Since this will be released as patch you need to create a template https://release.spryker.com/templates/pr-sdk?branch=Sdk.SyncApi.bugfix%2Fapps-4621-update-messages-shown-by-the-sync-api-sdk

**Review by stereomon — APPROVED** [review]

File: tests/_support/Helper/OpenApiValidatorHelper.php (hunk 5 above)
Comment (ghost) [reply tests/_support/Helper/OpenApiValidatorHelper.php:90]: In OpenApiValidateConsoleTest tests, we use haveDefaultOpenApiFile, which creates a folder structure according to the structure before specified in prepareOpenApiSchema. To check if the error text matches, we need to know in which folder the tested yml file is located. So, to keep the structure in one place it was decided to store the structure in a string and convert it to a multidimensional array if necessary.
